include/seqan3/io/sequence/sequence_file_format_fastq.hpp

```diff
@@ -0,0 +1,326 @@
+    sequence_file_format_fastq & operator=(sequence_file_format_fastq const &) = delete;
+    sequence_file_format_fastq(sequence_file_format_fastq &&) = default;
+    sequence_file_format_fastq & operator=(sequence_file_format_fastq &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "fastq" },
+        { "fq"    }
+    };
+
+    //!\copydoc sequence_file_in_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                                                   & stream,
+              sequence_file_in_options<seq_legal_alph_type, seq_qual> const & options,
+              seq_type                                                      & sequence,
+              id_type                                                       & id,
+              qual_type                                                     & qualities)
+    {
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+
+        auto stream_it = ranges::begin(stream_view);
+
+        // cache the begin position so we write quals to the same position as seq in seq_qual case
+        size_t sequence_size_before = 0;
+        size_t sequence_size_after = 0;
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size_before = ranges::size(sequence);
+
+        /* ID */
+        if (*stream_it != '@') // [[unlikely]]
+        {
+            throw parse_error{std::string{"Expected '@' on beginning of ID line, got: "} +
+                              detail::make_printable(*stream_it)};
+        }
+        ++stream_it; // skip '@'
+        if (options.truncate_ids)
+        {
+            ranges::copy(stream_view | ranges::view::take_while(!(is_cntrl || is_blank)),
+                         detail::make_conversion_output_iterator(id));
+            detail::consume(stream_view | view::take_line_or_throw);
+        }
+        else
+        {
+            ranges::copy(stream_view | view::take_line_or_throw,
+                         detail::make_conversion_output_iterator(id));
+        }
+
+        /* Sequence */
+        auto seq_view = stream_view | ranges::view::take_while(!is_char<'+'>{})    // until 2nd ID line TODO: or_throw
+                                    | ranges::view::remove_if(is_space);           // ignore whitespace
```

Comment:
in fasta we ignore also numbers...

Reply:
Yes, but AFAICT numbers are not legal for fastq